src/libraries/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorageFile.cs

```diff
@@ -734,7 +734,8 @@ public override void Remove()
 
             Close();
 
-            string parentDirectory = Path.GetDirectoryName(RootDirectory.TrimEnd(Path.DirectorySeparatorChar));
+            string? parentDirectory = Path.GetDirectoryName(RootDirectory.TrimEnd(Path.DirectorySeparatorChar));
+            Debug.Assert(parentDirectory != null);
```

Comment:
Why is this a given? If `RootDirectory` is something like `C:\` or `/` then `parentDirectory` can actually be null. Unless I'm missing other invariants in play here.

Reply:
Yes it can be null, and the usages below are not accepting null, that is why I am asserting here, might could catch the null reference if there is any in the future